### PR TITLE
SITL: added Glider simulator

### PR DESCRIPTION
**Pull request by tridge**
this adds a SITL model of this glider:
https://research.noaa.gov/2023/06/13/revolutionary-noaa-high-altitude-research-tool-passes-key-milestone/
this is the first of a series of PRs which will bring the firmware for this glider up-to-date with master

### CODE REVIEW

File: libraries/SITL/SIM_Glider.cpp [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+    filtered_servo_setup(2, 1100, 1900, model.elevatorDeflectionLimitDeg);
+    filtered_servo_setup(3, 1100, 1900, model.rudderDeflectionLimitDeg);
+    
+    float aileron  = 0.5*(filtered_servo_angle(input, 1) + filtered_servo_angle(input, 4));
+    float elevator = filtered_servo_angle(input, 2);
+    float rudder   = filtered_servo_angle(input, 3);
+    float balloon  = filtered_servo_range(input, 5);
+    float balloon_cut = filtered_servo_range(input, 9);
+
+    // Move balloon upwards using balloon velocity from channel 6
```
Comment (peterbarker) [inline libraries/SITL/SIM_Glider.cpp:217]: Not really sure what this comment means.  Stale?

File: libraries/SITL/SIM_Glider.cpp [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+    // update magnetic field
+    update_mag_field_bf();
+}
+
+/*
+   return true if we are on the ground
+*/
+bool Glider::on_ground() const
+{
+    // prevent bouncing around ground
```
Comment (peterbarker) [inline libraries/SITL/SIM_Glider.cpp:311]:
```suggestion
```

File: libraries/SITL/SIM_Glider.cpp [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+    // prevent bouncing around ground
+    // don't do ground interaction if being carried
+    return (hagl() <= 0.001f &&
+            carriage_state != carriageState::WAITING_FOR_PICKUP &&
+            carriage_state != carriageState::PRE_RELEASE &&
+            carriage_state != carriageState::WAITING_FOR_RELEASE);
```
Comment (peterbarker) [inline libraries/SITL/SIM_Glider.cpp:317]: `switch`

File: libraries/SITL/SIM_Glider.cpp [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+ */
+bool Glider::update_balloon(float balloon, Vector3f &force, Vector3f &rot_accel)
+{
+    if (!hal.util->get_soft_armed()) {
+        return false;
+    }
+
+    if (carriage_state != carriageState::WAITING_FOR_PICKUP &&
+        carriage_state != carriageState::PRE_RELEASE &&
+        carriage_state != carriageState::WAITING_FOR_RELEASE) {
```
Comment (peterbarker) [inline libraries/SITL/SIM_Glider.cpp:332]: `switch`

File: libraries/SITL/SIM_Glider.cpp [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+        // balloon not active
+        disable_origin_movement = false;
+        return false;
+    }
+
+    // while under balloon disable origin movement to allow for balloon position calculations
+    disable_origin_movement = true;
+
+    // assume a 50m tether with a 1Hz pogo frequency and damping ratio of 0.2
+    Vector3f tether_pos_bf = Vector3f(-1.0f,0.0f,0.0f); // tether attaches to vehicle tail approx 1m behind c.g.
```
Comment (peterbarker) [inline libraries/SITL/SIM_Glider.cpp:342]:
```suggestion
    Vector3f tether_pos_bf{-1.0f,0.0f,0.0f}; // tether attaches to vehicle tail approx 1m behind c.g.
```

File: libraries/SITL/SIM_Glider.cpp [hunk 6]
```diff
@@ -0,0 +1,403 @@
...
+
+        Vector3f tension_moment_vector_bf = tether_pos_bf % tension_force_vector_bf;
+        Vector3f tension_rot_accel = Vector3f(tension_moment_vector_bf.x/model.IXX, tension_moment_vector_bf.y/model.IYY, tension_moment_vector_bf.z/model.IZZ);
+        rot_accel = tension_rot_accel;
+
+        // add some rotation damping due to air resistance assuming a 2 sec damping time constant at SL density
+        // TODO model roll damping with more accuracy using Clp data for zero alpha as a first approximation
+        rot_accel -= gyro * 0.5 * air_density;
+    } else {
+        // thether is either slack awaiting pickup or released
```
Comment (peterbarker) [inline libraries/SITL/SIM_Glider.cpp:394]:
```suggestion
        // tether is either slack awaiting pickup or released
```

**Review by peterbarker — COMMENTED** [review]
Parameter metadata needed

File: libraries/SITL/SIM_Glider.cpp (hunk 1 above)
Comment (tridge) [reply libraries/SITL/SIM_Glider.cpp:217]: it does exactly what it says. servo channel 6 gives the balloon lift

File: libraries/SITL/SIM_Glider.cpp (hunk 2 above)
Comment (tridge) [reply libraries/SITL/SIM_Glider.cpp]: removed

File: libraries/SITL/SIM_Glider.cpp (hunk 3 above)
Comment (tridge) [reply libraries/SITL/SIM_Glider.cpp]: changed

File: libraries/SITL/SIM_Glider.cpp (hunk 4 above)
Comment (tridge) [reply libraries/SITL/SIM_Glider.cpp]: changed

File: libraries/SITL/SIM_Glider.cpp (hunk 5 above)
Comment (tridge) [reply libraries/SITL/SIM_Glider.cpp]: changed

File: libraries/SITL/SIM_Glider.cpp (hunk 6 above)
Comment (tridge) [reply libraries/SITL/SIM_Glider.cpp]: fixed

File: libraries/SITL/SIM_Glider.cpp [hunk 7]
```diff
@@ -0,0 +1,412 @@
...
+    // @Param: BLN_BRST
+    // @DisplayName: balloon burst height
+    // @Description: balloon burst height
+    // @Units: m
+    AP_GROUPINFO("BLN_BRST", 1, Glider, balloon_burst_amsl, 30000),
+
+    // @Param: BLN_RATE
+    // @DisplayName: balloon climb rate
+    // @Description: balloon climb rate
+    // @Units: m
```
Comment (IamPete1) [inline libraries/SITL/SIM_Glider.cpp:43]:
```suggestion
    // @Units: m/s
```

Comment (tridge) [reply libraries/SITL/SIM_Glider.cpp]: fixed

**Pull request merged**